### ISSUE TITLE
CA1304: Disallow null as culture

### ISSUE THREAD

**Issue by skrysmanski** [dotnet/roslyn-analyzers#6900]
### Analyzer

**Diagnostic ID**: [CA1304](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1304)

### Describe the improvement

The value `null` for the culture parameter should be disallowed by this rule.

For example, this rule is violated by the following code:

```c#
someInt.ToString("D");
```

Currently, a "clever" co-worker figured out they could fix the violation by passing `null`  as culture:

```c#
someInt.ToString("D", null);
```

In my opinion, this is not what the rule is for (as it doesn't increase the readability of the code) and the rule's description also doesn't read like this would be a correct/valid solution.

### Additional context

You should consider applying this change also to [CA1305](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1305).
